src/core/Akka.Persistence/Eventsourced.cs

```diff
@@ -52,10 +95,26 @@ public AsyncHandlerInvocation(object evt, Action<object> handler)
         }
 
         public object Event { get; }
-
         public Action<object> Handler { get; }
     }
 
+    /// <summary>
+    /// Non-stashing invocation with an asynchronous handler.
+    /// Used by <see cref="Eventsourced.PersistAsync{TEvent}(TEvent,Func{TEvent,Task})"/> and
+    /// <see cref="Eventsourced.DeferAsync{TEvent}(TEvent,Func{TEvent,Task})"/>.
+    /// </summary>
+    internal sealed class AsyncAsyncHandlerInvocation : IAsyncHandlerInvocation
```

Comment:
lol at the name:

1. 1 "async" for the type of handler
2. 1 "async" for the non-blocking effects of this